File: 1037 Valid Boomerang.py
class Solution:
    def isBoomerang(self, points: List[List[int]]) -> bool:
        a, b, c = points
        if a == b or b == c or c == a: return False
        return not (b[1] - a[1]) * (c[0] - a[0]) == (c[1] - a[1]) * (b[0] - a[0])
        

File: 1037 Valid Boomerang.java
class Solution {
    public boolean isBoomerang(int[][] points) {
        int[] a, b, c;
        a = points[0];
        b = points[1];
        c = points[2];
        if (a == b || b == c || c == a)
            return false;
        return !((b[1] - a[1]) * (c[0] - a[0]) == (c[1] - a[1]) * (b[0] - a[0]));
    }
}